crates/task-impls/src/consensus/helpers.rs

```diff
@@ -396,7 +396,10 @@ pub async fn publish_proposal_from_commitment_and_metadata<TYPES: NodeType>(
         view,
         quorum_membership,
         public_key.clone(),
-        Arc::clone(&consensus),
+        OuterConsensus::new(
+            "publish_proposal_from_commitment_and_metadata->parent_leaf_and_state",
```

Comment:
I think this change is awesome. My only concern is that if this method name changes, we lose the context in all the log spots and, as a result, the duplication of this name creates some cognitive load beyond what the editor could reasonably infer. What are your thoughts here? Perhaps a macro could be used to recover the name? I'm not sure if there's a better option.

Reply:
There's a crate that (allegedly) captures the call stack when acquiring a lock: https://crates.io/crates/rust-debugging-locks

Though I wasn't able to test it, but if we now have a wrapper struct used everywhere it might be easy to just swap out the underlying `RwLock` for the `RwLockWrapped` it provides and see if it works?

You might also be able to just capture/print a `Backtrace`: https://doc.rust-lang.org/std/backtrace/struct.Backtrace.html (though I don't really know if that does what I think it does)

Reply:
I also think we should look into what the tracing library can do as well.  Specifically if we set up our tracing spans properly this should give us a good idea of who is doing the locking